pkg/credentialprovider/provider.go

```diff
@@ -83,7 +83,7 @@ func (d *defaultDockerConfigProvider) Enabled() bool {
 // Provide implements dockerConfigProvider
 func (d *defaultDockerConfigProvider) Provide() DockerConfig {
 	// Read the standard Docker credentials from .dockercfg
-	if cfg, err := ReadDockerConfigFile(); err == nil {
+	if cfg, err := ReadDockerConfigFile(nil); err == nil {
```

Comment:
I don't see why you wouldn't want to accept a set of paths for both and say "find a valid config.json or .dockercfg somewhere in here"